### ISSUE TITLE
Manually created tickets should show the agent's name in the initial post

### ISSUE THREAD

**Issue by joeldeteves** [osTicket/osTicket#5807]
### Prerequisites

* [ Y ] Can you reproduce the problem in a fresh installation of the "develop" branch?
* [ N ] Do you have any errors in the PHP error log, or javascript console?
* [ Y ] Did you check the [osTicket forums](https://forum.osticket.com/)?
* [ Y ] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

### Description

Cross posting from the OST forum per suggestion from @ntozier: https://forum.osticket.com/d/98994-manually-created-tickets-should-show-the-agent-s-name

> Hello, this is a suggestion to modify the behavior of how OSTicket currently handles the description of tickets that are created manually in the agent panel by an agent.
> 
> Currently, when an agent creates a ticket and assigns it to a user, the description of the notes reads as "user posted" followed by the timestamp, e.g. if the user's name is "Bob Smith" the initial note on the ticket says "Bob Smith posted 02/12/2021 6:49 PM", even though it was the agent who created the ticket.
> 
> The problem is, when an agent is the one who created the ticket this winds up being somewhat misleading for tracking purposes as it can wind up leading the user to believe the system posted something they did not say, and if ever recalled in a dispute scenario it can cause communications issues about who said what.
> 
> As ridiculous as this sounds, being in the tech space I am sure I am not the only one who has had to pull up "historical information" to settle a "he said she said" situation with end users (as much as we generally strive to have great relationships with our clients).
> 
> I believe a better solution would be, when an agent creates a ticket manually in the ticket system it should show the agent's name under the initial post and only show the user's name when they actually reply to the ticket (or when the user is the one who opened the ticket).
> 
> Thank you,

### Steps to Reproduce

1. Create a new ticket in the control panel
2. Assign the ticket to a user
3. Enter the ticket details

**Expected behavior:** Ticket description when created by an agent should read "*Agent name posted*"

**Actual behavior:** Ticket description when created by an agent shows "*User name posted*"

### Versions
OsTicket 1.15.1
PHP 7.4.13
NGINX 1.18.0

(Issue is not browser or OS related)

**Comment by nachosavage** [comment]
+1

**Comment by smassonnat** [comment]
+1

**Comment by ntozier** [comment]
Manually created tickets already have the name of the agent who created them on them.

![2021-08-16_09-14-33](https://user-images.githubusercontent.com/1315768/129569915-57499085-ee18-4376-a278-4a7e87479907.png)

**Comment by joeldeteves** [comment]
> Manually created tickets already have the name of the agent who created them on them.
> 
> ![2021-08-16_09-14-33](https://user-images.githubusercontent.com/1315768/129569915-57499085-ee18-4376-a278-4a7e87479907.png)

I’ve updated the description of this issue. It’s not the “created by” that’s the problem - its the initial reply (see my steps on how to reproduce).

**Comment by ntozier** [comment]
Ohhh gotcha!  :)  Thanks for the clarification.

**Comment by nachosavage** [comment]
yes, precisely.

we are trying to solve the problem by having a "user" that we asigne agent created tickets to (something like "agent created ticket") and the ad the corresponding client as a collaborator. 

that helps tracking all the agent created tickets, but i haven't check how searches act on it (if searching by company name would work with the collaborator for example)

**Comment by Zahrun** [comment]
@joeldeteves 

As per https://github.com/osTicket/osTicket/pull/6193#issuecomment-1183257292:
> You can make a Help Topic specific for these types of Tickets and then go to the Topic in Admin Panel > Manage > Help Topics, click the Topic, click Forms tab, then disable (uncheck) the Issue Details field, and Save Changes. Now when Agents create a Ticket with that Topic they will not be able to input an initial Message.

That way the agent can fill the initial response, which will show under their name and the initial message by the user is simply not there.
If that solution is fine for you, you may close this issue.

**Comment by joeldeteves** [comment]
Hi @Zahrun not really a good workaround for us - we have more than a dozen help topics and having to use a special topic just to create a ticket and then switch topics afterwards is a lot of extra work for the ticket volume we deal with.

Plus it doesn't follow good design principles IMO - if an agent creates a ticket then it should say in the audit trail that the agent created the initial reply.

Every other ticket system out there does this.

My 2 cents

**Comment by Zahrun** [comment]
Ok let’s see if we can have it as a setting

**Comment by JediKev** [comment]
@joeldeteves 

It does say that the Agent created the Ticket in the Thread Event. The initial message can be ignored, just focus on the thread event for ticket creation. Because as I mentioned in the other thread, the Agent is creating the Ticket _on behalf of_ the User so the initial message should say it's "from the User".

Cheers.

**Comment by joeldeteves** [comment]
Sorry @JediKev - getting mixed up again between ticket created and initial reply.

The issue is the initial reply (fixed my last comment) - it is a matter of the user didn't actually post the reply so it becomes an issue of "he said / she said"

**Comment by JediKev** [comment]
@joeldeteves 

We think removing the restrictions on Visible/Required/Editable for Agents for the Issue Details field would be best. This way you can make it non-required for Agents so they don't have to input an initial message for any Help Topic.

Cheers.

**Comment by joeldeteves** [comment]
Hi @JediKev the problem with that approach is we often use that field to enter a detailed description of the issue the customer is facing.

So I'm not sure removing it entirely for agents would be the right solution

**Comment by JediKev** [comment]
@joeldeteves 

It won't remove it, it will just make it non-required.

Cheers.

**Comment by joeldeteves** [comment]
Yea but ignoring the field altogether is fundamentally the same as "removing" it for our use case.

Say we have a phone person (as tech support companies often do).

She takes the call from the customer and creates a ticket for them.

Where should she enter the details of the request?

**Comment by protich** [comment]
@joeldeteves - I think we can all agree that the usecase vary widely... you use osTicket instead of other ticket systems with "good design priciples" for a reason. 

The last use case you mentioned is _exactly_ the reason of having the initial message field - posted on behalf of the user!  Yes, it can look a little confusing, as you mentioned, but it was **important to clearly mark it as a message / issue / report from the end user**... in keeping with the color coding scheme we use on Messages, Replies and Internal Notes within the ticket thread which is tied to Thread Entry type.

Here is a little background... the reason we couldn't use the User's name is more historical and technical than simple visual fix. Initial Thread Entry has always been message from End User - in fact osTicket in the olden days required it for ticket thread to load. We ended up making it optional via Help Topic setting as Kevin mentioned to cover different use cases. On technical side, osTicket has separate tables for Agents (staff) and End Users (user) - a Thread Entry of type "M" (Message) uses `user_id` instead of `staff_id` to display Poster information. 

Could we also populate `staff_id` and visually credit the Agent on behalf of the User?  **Absolutely** - but this is a feature request as opposed to a bug report. Perhaps it's something you can help with by making  a pull request?

**Comment by joeldeteves** [comment]
Thank you for the detailed explanation @protich - I believe @Zahrun is working on a pull request already - perhaps it can be modified to fit your suggestion.

**Comment by Zahrun** [comment]
@protich, please see and comment https://github.com/osTicket/osTicket/pull/6193

**Comment by protich** [comment]
The pull request linked is hacky - it clears user information. Also Agent's name can change - so saving `staff_id` in the thread entry is better solution.... poster should be something like  **Agent's Name (for User's Name)** making it clear it's message posted onbehalf of the user.

**Comment by Zahrun** [comment]
If we all agree on the "Agent's Name (for User's Name)" specification, I’ll change the PR to have that in a less hacky way.

**Comment by JediKev** [comment]
@Zahrun 

Thank you for the reply but there is no need; we have a pull for this coming soon.

Cheers.

**Comment by mbassiouny33** [comment]
Hey, another +1 for this. We have a use case where all tickets are created by "agents" and are for internal usage, so we don't care about the "from [client name]" but the person who opened the ticket. Could be great if the client name can be left empty and just count on "created by/agent name" in our use-case.

Has this been addressed in the recent versions of osticket? I see https://github.com/osTicket/osTicket/pull/6193 is still open

**Comment by joeldeteves** [comment]
I can see this is still the ongoing behavior in 2024, was there ever any progress made on this?

**Comment by JediKev** [comment]
@joeldeteves 

No, no progress at this time (as you can tell by this thread). Legacy (current codebase) is in maintenance mode so it will not receive any major new features, etc. whilst we focus on v2.0. There was a pull referenced above however we didn't like it. We were working on a pull ourselves but v2.0 got in the way (priority). So for now you will either need to modify the code yourself or wait until a pull is made.

Cheers.

**Comment by joeldeteves** [comment]
Okay thank you - will this be part of OST 2.0 then?

**Comment by JediKev** [comment]
@joeldeteves 

Probably not off-the-bat but it's on our Feature Request list; so it's _slated_ for v2.0.

Cheers.

**Comment by joeldeteves** [comment]
> @joeldeteves
> 
> Probably not off-the-bat but it's on our Feature Request list; so it's _slated_ for v2.0.
> 
> Cheers.

Thank you! I'm happy to wait it out, been looking forward to 2.0 for a long time
